ckan/lib/mailer.py

```diff
@@ -141,7 +141,8 @@ def get_reset_link(user):
                    h.url_for(controller='user',
                              action='perform_reset',
                              id=user.id,
-                             key=user.reset_key))
+                             key=user.reset_key,
+                             qualified=True))
```

Comment:
I'm afraid I don't understand. On my CKAN I went from a missing site_url in the link to having one, so it appears to work. I was only implementing the suggested solution and trying to feed back as a pull request.